### PR TITLE
Script completion proposal

### PR DESCRIPTION
**Pull request by williamthome**
### Description

Adds completion to `bin/zotonic`.
Running `make` or `bin/zotonic completion` a file named `zotonic-completion.bash` is created in `/etc/bash_completion.d`.
No completion for options, just for the `zotonic_cmd_*` commands.
Working in Ubuntu 21.10. Would this work in other OS?

zotonic_completion

The generated file `zotonic-completion.bash` content:
```bash
#/usr/bin/env bash

_zotonic_completions()
{
	if [ "${#COMP_WORDS[@]}" != "2" ]; then
		return
	fi

	COMPREPLY=($(compgen -W "addsite compile compilefile config configfiles configtest connectdb createdb debug dispatch etop flush load logtail restart restartsite rpc runtests shell siteconfig siteconfigfiles sitedir sitetest start start_nodaemon startsite status stop stopsite update wait" "${COMP_WORDS[1]}"))
}

complete -F _zotonic_completions zotonic
```

### Checklist

- [ ] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

File: apps/zotonic_launcher/bin/zotonic [hunk 1]
```diff
@@ -17,6 +17,15 @@ export ZOTONIC_BIN
 cd -- "${ZOTONIC}" || \exit 1
 
 case "$1" in
+    completion)
+        COMMAND_NAMES="$(${ZOTONIC_BIN}/zotonic.escript command_names)"
+        # TODO: Check OS (this works in Ubuntu 21.10)
+        SYS_COMPLETION_DIR="/etc/bash_completion.d"
+        OUTPUT="$SYS_COMPLETION_DIR/zotonic-completion.bash"
+        # TODO: Completion for options
+        sudo sh -c "printf '#/usr/bin/env bash\n\n_zotonic_completions()\n{\n\tif [ \"\${#COMP_WORDS[@]}\" != \"2\" ]; then\n\t\treturn\n\tfi\n\n\tCOMPREPLY=(\$(compgen -W %s \"\${COMP_WORDS[1]}\"))\n}\n\ncomplete -F _zotonic_completions zotonic' '$COMMAND_NAMES' >| '$OUTPUT'"
```
Comment (vkatsuba) [inline apps/zotonic_launcher/bin/zotonic:26]: GitHub Actions provide error:
```sh
 bin/zotonic completion
./zotonic: 26: ./zotonic: sudo: not found
./zotonic: 27: ./zotonic: sudo: not found
```
I suppose `sudo` should not used in similar scripts.

Comment (vkatsuba) [inline apps/zotonic_launcher/bin/zotonic:26]: GitHub Actions provide error:
``sh
 bin/zotonic completion
./zotonic: 26: ./zotonic: sudo: not found
./zotonic: 27: ./zotonic: sudo: not found
```
Looks like `sudo` command should not need to use in similar scripts.

**Comment by mworrell** [comment]
Does not work on macOS, but also no errors, so I am ok with merging this for Linux users (and maybe more *nix or BSD)

@mmzeeman agree?

**Comment by mworrell** [comment]
Thanks!

**Comment by williamthome** [comment]
Sorry, this really doesn't work. Worked for me because I have the completion file at '/etc/bash_completion.d/' of the first implementation. I have the solution and will open a new PR tonight to fix this.

**Pull request merged**